dali/operators/reader/webdataset_reader_op.cc

```diff
@@ -0,0 +1,94 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/webdataset_reader_op.h"
+#include <algorithm>
+#include <cstring>
+#include <string>
+
+namespace dali {
+
+void WebdatasetReader::RunImpl(HostWorkspace& ws) {
+  int num_outputs = ws.NumOutput();
+  int num_samples = GetCurrBatchSize();
+
+  for (int data_idx = 0; data_idx < num_samples; data_idx++) {
+    auto& sample = GetSample(data_idx);
+    for (int output_idx = 0; output_idx < num_outputs; output_idx++) {
+      ws.OutputRef<CPUBackend>(output_idx)[data_idx].Resize(sample[output_idx].shape());
+      ws.OutputRef<CPUBackend>(output_idx)[data_idx].set_type(sample[output_idx].type());
+      ws.OutputRef<CPUBackend>(output_idx)[data_idx].SetMeta(sample[output_idx].GetMeta());
+      std::memcpy(ws.OutputRef<CPUBackend>(output_idx)[data_idx].raw_mutable_data(),
+                  sample[output_idx].raw_data(), sample[output_idx].nbytes());
+    }
+  }
+}
+
+DALI_SCHEMA(readers__Webdataset)
+    .DocStr(
+        R"code(A reader for the webdataset format.
+
+The webdataset format is a dataset meant to improve caching of the data between the samples read.
+The data itself is contained within one or several tar archives, each of which is further split into
+samples, each of which contains one or several components that correspond to the actual file
+contained within the archive. The components that correspond to specific sample are aggregated by
+the part of the filepath that does not correspond to the extension (for the specifics about the
+extensions please read the description of the ``ext`` parameter below). Note that samples with
+their filename starting with a dot will not be loaded, as well as entries that are not a file.
+
+In addition to the tar archive with data, each archive should come with a corresponding index file.
+The format of the index file is as follows:
+<offset behind the contents of the last archive entry> <number_of_samples>
+<sample1_start_offset> <component1_ext> <component1_size> <component2_ext> <component2_size> ...
+<sample2_start_offset> <component1_ext> <component1_size> <component2_ext> <component2_size> ...
+...)code")
+    .NumInput(0)
+    .OutputFn([](const OpSpec& spec) {
+      return spec.HasArgument("ext") ? spec.GetRepeatedArgument<std::string>("ext").size() : 0;
+    })
+    .AddArg("uris", R"code(The list of (one or more) paths to the webdataset archives.
+Has to be the same length as the ``configs`` argument.)code",
+            DALI_STRING_VEC)
+    .AddArg("configs",
+            R"code(The list of the index files corresponding to the respective webdataset archives.
+Has to be the same length as the ``uris`` argument.)code",
```

Comment:
```suggestion

Has to be the same length as the ``uris`` argument.)code",
```

Reply:
This empty line is important. It makes the next line appear in the new line, otherwise, it will be the same as the preceding sentence. The guide is to have the first sentence in a separate line that the following one.

Reply:
done